xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/BasePage.java

```diff
@@ -118,6 +125,38 @@ public BasePage()
     {
         super();
         waitUntilPageIsReady();
+
+        Logger LOGGER = LoggerFactory.getLogger(BasePage.class);
+        WCAGContext wcagContext = getUtil().getWcagContext();
+        if (!wcagContext.isWcagEnabled()) {
+            return;
+        }
+        if (wcagContext.checkAccessibility(this.getPageURL(), this.getClass())) {
+            /* Perform accessibility checks on the element when instanciated if the wcag is activated in build. */
+            long startTime = System.currentTimeMillis();
+            XWikiWebDriver driver = this.getDriver();
+            try {
+                if (driver == null) {
```

Comment:
can this happen?

Reply:
I shouldn't happen, since the getDriver returns a driver that's always initialized during construction of PersistentTestContext. The instantiation of the driver happens either in the PersistentTestContext constructor or beforehand.
The driver being null would surely make the test framework fail before reaching this point anyways.

I'm removing this clause for the next commit :+1: